pkg/registry/core/pod/storage/eviction.go

```diff
@@ -234,8 +231,10 @@ func (r *EvictionREST) checkAndDecrement(namespace string, podName string, pdb p
 
 	if pdb.Status.DisruptedPods == nil {
 		pdb.Status.DisruptedPods = make(map[string]metav1.Time)
+	} else if _, ok := pdb.Status.DisruptedPods[podName]; ok {
```

Comment:
I think we still want to update the timestamp in the DisruptedPods map even if the pod has already been counted in PodDisruptionsAllowed. Otherwise there is a chance the entry in the map might time out too early.